From: Jakub Jirutka <jakub@jirutka.cz>
Date: Fri, 6 May 2016 00:09:09 +0200
Subject: [PATCH] Fix missing include sys/types.h

Without this patch compilation on musl libc fails with:

    unknown type name 'u_int32_t'
    unknown type name 'u_int64_t'
---
 SftpServer/Encode.c | 1 +
 SftpServer/Handle.c | 1 +
 SftpState/Main.c    | 1 +
 SftpWho/Main.c      | 1 +
 4 files changed, 4 insertions(+)

diff --git a/SftpServer/Encode.c b/SftpServer/Encode.c
index a41c630..31a6b90 100644
--- a/SftpServer/Encode.c
+++ b/SftpServer/Encode.c
@@ -20,6 +20,7 @@
 #include "../config.h"
 #include <errno.h>
 #include <sys/ioctl.h>
+#include <sys/types.h>
 #include <grp.h>
 #include <pwd.h>
 #include <stdlib.h>
diff --git a/SftpServer/Handle.c b/SftpServer/Handle.c
index 5f795ac..2f367be 100644
--- a/SftpServer/Handle.c
+++ b/SftpServer/Handle.c
@@ -18,6 +18,7 @@
  */
 
 #include "../config.h"
+#include <sys/types.h>
 #include <errno.h>
 #include <stdio.h>
 #include <stdlib.h>
diff --git a/SftpState/Main.c b/SftpState/Main.c
index 0e9398f..70cbb88 100644
--- a/SftpState/Main.c
+++ b/SftpState/Main.c
@@ -18,6 +18,7 @@ Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA  02111-1307, USA.
 */
 
 #include "../config.h"
+#include <sys/types.h>
 #include <errno.h>
 #include <fcntl.h>
 #include <stdio.h>
diff --git a/SftpWho/Main.c b/SftpWho/Main.c
index 455da4a..25dcbf3 100644
--- a/SftpWho/Main.c
+++ b/SftpWho/Main.c
@@ -19,6 +19,7 @@ Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA  02111-1307, USA.
 */
 
 #include "../config.h"
+#include <sys/types.h>
 #include <stdio.h>
 #include <time.h>
 #include <stdlib.h>
-- 
2.8.1

